### PR TITLE
[To Metrics branch] Enable Exemplars for Histogram

### PR DESCRIPTION
**Pull request by cijothomas**
Sending a PR with a testable prototype  for Exemplars:
1. Hard coded AlignedHistogramBucketExemplarReservoir, enabled for all Histograms with buckets.
2. Exemplar just stores the ts, value, TraceId, SpanId (this should be sufficient to test the basic capabilities in prometheus -> tracebackends)
3. Console and OTLPExporter exports Exemplars, if available.

Will send a PR with examples on how to run this to experience Exemplars E2E, using OTel .NET + OTLP + Prometheus

Note: 
This is to show a quick E2E exemplar demo.
ExemplarFilters are not integrated.
Only hardcoded for Histograms, with no ability to turn off.

Subsequent PRs will address the issues one by one.

### CODE REVIEW

File: src/OpenTelemetry.Exporter.Console/ConsoleMetricExporter.cs [hunk 1]
```diff
@@ -167,6 +167,23 @@ public override ExportResult Export(in Batch<Metric> batch)
...
 
+                    var exemplarString = new StringBuilder();
+                    foreach (var exemplar in metricPoint.GetExemplars())
+                    {
+                        if (exemplar.Timestamp != default)
+                        {
+                            exemplarString.Append("value: ");
+                            exemplarString.Append(exemplar.DoubleValue);
+                            exemplarString.Append(" Timestamp: ");
+                            exemplarString.Append(exemplar.Timestamp);
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.Console/ConsoleMetricExporter.cs:178]: What's the culture/format?

File: src/OpenTelemetry.Exporter.Console/ConsoleMetricExporter.cs [hunk 2]
```diff
@@ -182,6 +199,13 @@ public override ExportResult Export(in Batch<Metric> batch)
                     msg.Append(metric.MetricType);
                     msg.AppendLine();
                     msg.Append($"Value: {valueDisplay}");
+
+                    if (exemplarString.Length > 0)
+                    {
+                        msg.AppendLine();
+                        msg.Append("Exemplars \n");
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.Console/ConsoleMetricExporter.cs:206]: Curious, why is ` ` needed?

Comment (reyang) [reply src/OpenTelemetry.Exporter.Console/ConsoleMetricExporter.cs:206]: `\n` -> AppendLine

File: src/OpenTelemetry.Exporter.Console/ConsoleMetricExporter.cs (hunk 1 above)
Comment (cijothomas) [reply src/OpenTelemetry.Exporter.Console/ConsoleMetricExporter.cs]: made to be similar to MP's timestampt

**Review by reyang — APPROVED** [review]

File: src/OpenTelemetry.Exporter.Console/ConsoleMetricExporter.cs [hunk 3]
```diff
@@ -167,6 +167,23 @@ public override ExportResult Export(in Batch<Metric> batch)
                         }
                     }
 
+                    var exemplarString = new StringBuilder();
+                    foreach (var exemplar in metricPoint.GetExemplars())
+                    {
+                        if (exemplar.Timestamp != default)
+                        {
+                            exemplarString.Append("Value: ");
```
[the remark this answers is not in the document]
Comment (cijothomas) [reply src/OpenTelemetry.Exporter.Console/ConsoleMetricExporter.cs:175]: yup! The prometheus output shared in the PR looks very nice :) @Yun-Ting lets sync and see if this is something you want to attempt to do in console exporter.

**Pull request merged**